browser-sign-in/app/src/main/java/com/okta/android/samples/browser_sign_in/util/FingerprintDialog.java

```diff
@@ -0,0 +1,256 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.okta.android.samples.browser_sign_in.util;
+
+import android.animation.Animator;
+import android.annotation.TargetApi;
+import android.app.DialogFragment;
+import android.app.FragmentTransaction;
+import android.content.Context;
+import android.graphics.Color;
+import android.hardware.fingerprint.FingerprintManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.CancellationSignal;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.animation.DecelerateInterpolator;
+import android.view.animation.OvershootInterpolator;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.okta.android.samples.browser_sign_in.R;
+import com.okta.oidc.storage.security.EncryptionManager;
+
+import java.lang.ref.WeakReference;
+
+@TargetApi(23)
+public class FingerprintDialog extends DialogFragment {
+    public static final int ANIMATION_DURATION = 500;
+
+    protected TextView mTextViewStatus;
+    protected ImageView mImageViewStatus;
+
+    private FingerprintManager mFingerprintManager;
+    private CancellationSignal mCancellationSignal;
+    private FingerprintManager.CryptoObject mCryptoObject;
+    private Context mContext;
+    private int mPurpose;
+    private WeakReference<FingerprintDialogCallbacks> mFingerprintDialogCallbacks;
+
+    public static FingerprintDialog newInstance() {
+        return new FingerprintDialog();
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Do not create a new Fragment when the Activity is re-created such as orientation changes.
+        setRetainInstance(true);
+
+        mContext = getActivity().getApplicationContext();
+        mFingerprintManager = (FingerprintManager)
+                mContext.getSystemService(Context.FINGERPRINT_SERVICE);
+        this.setCancelable(false);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        mCancellationSignal = new CancellationSignal();
+
+            try {
+                // Start listening for fingerprint events
+                mFingerprintManager.authenticate(mCryptoObject, mCancellationSignal,
+                        0, new AuthCallbacks(), null);
+            } catch (IllegalArgumentException | IllegalStateException | SecurityException e) {
+                // Should never be thrown since we have declared the USE_FINGERPRINT permission
+                // in the manifest
+            }
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+
+        // If the fingerprint authentication process is running, cancel it.
+        mCancellationSignal.cancel();
```

Comment:
Is this being called? Because the fingerprint scanner is still working even without the dialog showing. I can simulate a scan first then press refresh and it won't prompt me for authentication.